### PR TITLE
Deprecate CoreCH, fallback to using CH

### PR DESCRIPTION
**Pull request by karenzshea**
# Issue

Closes https://github.com/Project-OSRM/osrm-backend/issues/4444 and https://github.com/Project-OSRM/osrm-backend/issues/4482

## Tasklist
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add changelog entry
 - [ ] add regression / cucumber cases (see docs/testing.md) how to test this??
 - [ ] review
 - [ ] adjust for comments

## Requirements / Relations
cc @TheMarex @oxidase

### CODE REVIEW

File: src/engine/routing_algorithms/direct_shortest_path.cpp [hunk 1]
```diff
@@ -65,11 +65,6 @@ InternalRouteResult directShortestPathSearch(SearchEngineData<Algorithm> &engine
 }
 
 template InternalRouteResult
-directShortestPathSearch(SearchEngineData<corech::Algorithm> &engine_working_data,
-                         const DataFacade<corech::Algorithm> &facade,
-                         const PhantomNodes &phantom_nodes);
-
-template InternalRouteResult
 directShortestPathSearch(SearchEngineData<ch::Algorithm> &engine_working_data,
```
Comment (oxidase) [inline src/engine/routing_algorithms/direct_shortest_path.cpp]: can also be removed but requires a definition change above to
```
template <>
InternalRouteResult directShortestPathSearch(SearchEngineData<ch::Algorithm> &engine_working_data,
                                             const DataFacade<ch::Algorithm> &facade,
                                             const PhantomNodes &phantom_nodes)
```

File: src/tools/contract.cpp [hunk 2]
```diff
@@ -49,7 +49,7 @@ return_code parseArguments(int argc,
         "Number of threads to use")(
         "core,k",
         boost::program_options::value<double>(&contractor_config.core_factor)->default_value(1.0),
-        "Percentage of the graph (in vertices) to contract [0..1]")(
+        "Percentage of the graph (in vertices) to contract [0..1]. Will always be 1.0")(
```
Comment (oxidase) [inline src/tools/contract.cpp]: would  be `DEPRECATED Percentage of the graph (in vertices) to contract [0..1]` better?

File: include/contractor/contractor_config.hpp [hunk 3]
```diff
@@ -65,6 +65,7 @@ struct ContractorConfig final : storage::IOConfig
 
     unsigned requested_num_threads;
 
+    // Deprecated
```
Comment (oxidase) [inline include/contractor/contractor_config.hpp]: Please could you add also a release version when it is planed to remove, like `// DEPRECATED to be removed in v6.0`

**Comment by oxidase** [comment]
as per chat with @MoKob: all internal usage of `core_factor` can be removed 
* https://github.com/Project-OSRM/osrm-backend/blob/e647547adde8239ab637d1a443263562e1e2d34f/src/contractor/graph_contractor.cpp#L564 and all dependencies removed or adjusted using `core_factor=1`

File: include/contractor/contractor_config.hpp (hunk 3 above)
Comment (karenzshea) [reply include/contractor/contractor_config.hpp]: I'll open a follow up issue to track removal in v6 too

File: src/tools/contract.cpp (hunk 2 above)
Comment (karenzshea) [reply src/tools/contract.cpp]: done in https://github.com/Project-OSRM/osrm-backend/pull/4601/files#diff-097d6382cc3c7256f51a139534b8c76dR52

**Pull request closed without merging**